feat(std/help): Add `--help` for external-commands

Tyarel8 opened the pull request:
# Description
I have just discovered the `std/help` command and that it can use `man` or other programs for externals. Coming from windows, I don't have `man` so what I want is just to run `external_program --help` in most cases. This pr adds that option, if you set `$env.NU_HELPER = "--help"`, it will run the command you passed with `--help` added as the last argument.

![image](https://github.com/user-attachments/assets/60d25dda-718b-4cb5-b540-808de000b221)

# User-Facing Changes
None

# Tests + Formatting


# After Submitting

132ikl commented:
I don't think this is a great idea because it could cause issues for commands that don't have a `--help`, which might cause unexpected issues. For example, doing `help discord` would cause Discord to launch which definitely isn't what you would want that to do

Tyarel8 commented:
> Maybe we could put that in a try/catch and if it fails, in the catch use `/?` or `-h` just to make it more thorough?

I don't know if that is a good idea, like the discord example earlier commands that don't have a `--help` option won't necessarily fail and trying more think could make it even worse.

Tyarel8 commented:
> wherever this is documented we need to note that it might cause issues

Very true, I have added to the `help help` description the documentation for how `$env.NU_HELPER` works in general and added a warning for using the new `--help` option.

Tyarel8 commented:
Also added that the external helper removes the `^` character, so if you want to get the help of an external with the same name as an internal you can do so with `help ^ls`

![image](https://github.com/user-attachments/assets/103f348e-b553-4b18-b4ea-bfc372f2ba5b)
